src/libsystemd/sd-event/sd-event.c

```diff
@@ -1494,6 +1499,22 @@ _public_ int sd_event_source_set_io_fd(sd_event_source *s, int fd) {
         return 0;
 }
 
+_public_ int sd_event_source_get_io_fd_own(sd_event_source *s, int *own) {
+        assert_return(s, -EINVAL);
+        assert_return(s->type == SOURCE_IO, -EDOM);
+
+        *own = s->io.owned;
```

Comment:
ah, sorry, one more change please that i missed earlier: when returning booleans in our getter calls we so far just returned this in the return value, see `sd_event_get_watchdog()` as one example. i.e. negative returns indicate erorr, 0 means "false" and > 0 means "true".

Hence, please drop the extra argument and just do `return s->io.owned`